src/patterns/ec2-app.ts

```diff
@@ -5,21 +5,48 @@ import type { GuCertificateProps } from "../constructs/acm";
 import { GuCertificate } from "../constructs/acm";
 import type { GuUserDataProps } from "../constructs/autoscaling";
 import { GuAutoScalingGroup, GuUserData } from "../constructs/autoscaling";
-import { Gu5xxPercentageAlarm } from "../constructs/cloudwatch";
 import type { Gu5xxPercentageMonitoringProps, NoMonitoring } from "../constructs/cloudwatch";
+import { Gu5xxPercentageAlarm } from "../constructs/cloudwatch";
 import type { GuStack } from "../constructs/core";
 import { AppIdentity } from "../constructs/core/identity";
-import { GuVpc, SubnetType } from "../constructs/ec2";
+import { GuSecurityGroup, GuVpc, SubnetType } from "../constructs/ec2";
 import { GuGetPrivateConfigPolicy, GuInstanceRole } from "../constructs/iam";
 import {
   GuApplicationListener,
   GuApplicationLoadBalancer,
   GuApplicationTargetGroup,
 } from "../constructs/loadbalancing";
+import { transformToSecurityGroupAccessRule } from "../utils/security-groups";
+
+const PUBLIC = "PUBLIC" as const;
+const INTERNAL = "INTERNAL" as const;
+const RESTRICTED = "RESTRICTED" as const;
+
+interface Access {
+  type: string;
+}
+
+// Everyone on the internet can access
+interface PublicAccess extends Access {
+  type: typeof PUBLIC;
+}
+
+// Access restricted to internal AWS networks
+interface InternalAccess extends Access {
+  type: typeof INTERNAL;
+}
+
+// Access whitelisted to the CIDR ranges passed in
+interface RestrictedAccess extends Access {
+  type: typeof RESTRICTED;
+  cidrRanges: string[];
+}
+
+type AppAccess = PublicAccess | InternalAccess | RestrictedAccess;
```

Comment:
Perhaps we should add a type which implements `RestrictedAccess` with CIDR ranges for all Guardian networks, as I guess this is the restricted access that most people would need?

Reply:
That sounds good, although that would need access to @guardian/private-infrastructure-config, which I believe would break CI as we don't have the appropriate SSH access set up in GitHub Actions, I _think_?